docs/usage.md

```diff
@@ -70,6 +45,76 @@ INIT_RUNTIME(2); // This is the number of pkeys needed
 
 Note that this must only be included in one source file per DSO.
 
+### Capturing Compile Commands
+
+The rewriter relies on a `compile_commands.json` file with information about the
+compiler flags used to compile each source file. If your build system does not
+directly support generating a compile commands file, you can use a tool like
+`bear` to record the compiler invocations that your build system makes.
+
+One limitiation is that the rewriter currently does not support having multiple
+compile commands for the same source file. If your build system builds the same
+source file multiple times, e.g. building for both a shared lib and a static
+lib, you'll need to change your build such that it only builds the sources once.
```

Comment:
It "supports" this, in that it'll just warn and choose the first one.

Reply:
Oh good point. The warnings are pretty easy to miss in the rewriter's output, so I think I still want the warning here, but I'll reword it to be more accurate.